File: week-04/day-01/Animal.py
class Animal(object):
    def __init__(self, hunger_value, thirst_value):
        self.hunger_value = int(hunger_value)
        self.thirst_value = int(thirst_value)

    def eat(self):
        self.hunger_value -= 1

    def drink(self):
        self.thirst_value -= 1

    def play(self):
        self.hunger_value += 1
        self.thirst_value += 1

zebra = Animal(10, 10)
lion = Animal(20, 20)

print(zebra.thirst_value)
zebra.drink()
print(zebra.thirst_value)

print(lion.hunger_value)
lion.play()
print(lion.hunger_value)

File: week-08/Animal.js
function Animal(type) {
  this.type = type,
  this.hunger = 5,
  this.thirst = 5,
  this.eat = function() {
    if (this.hunger !== 0) {
      this.hunger -= 1;
    }
  },
  this.drink = function() {
    if (this.thirst !== 0) {
      this.thirst -= 1;
    }
  },
  play = function() {
    if (this.hunger !== 0 && this.thirst !== 0) {
      this.hunger += 1;
    this.thirst += 1;
    }
  }
}

function Farm(slots) {
  this.slots = slots,
  this.animals = [],

  this.breed = function() {
    if (this.slots !== 0) {
      let Sheep = new Animal('sheep');
      this.animals.push(Sheep);
      this.slots -= 1;
    }

  this.slaughter = function() {
    let minHunIndex = 0;
    for (i = 0; i <= animals; i++) {
      if (animals[i].hunger < minHunIndex) {
        minHunIndex = i;
      }
      }
      this.animals.splice(minHunIndex, 1);
      }
    }
}

const SheepFarm = new Farm(20);


console.log(SheepFarm.animals); // Should log 20 Animal objects

// const button = document.querySelector('button');

// Add a click event to the button and call 'progress'

// The progress function should log the following to the console:
//  - The farm has 20 living animals, we are full